Runtime/Interaction/InteractableObject.cs

```diff
@@ -87,11 +87,29 @@ public bool IsUsable
         protected override void Reset()
         {
             base.Reset();
-            
+
             // Sets the 'interactionLayerMask' to Default in order to not interact with Teleportation or UI rays.
             interactionLayerMask = 1;
         }
 
+        internal void OnTriggerEnter(Collider other)
+        {
+            SnapZone target = other.gameObject.GetComponent<SnapZone>();            
+            if (target != null && target.enabled && !IsInSocket)
```

Comment:
Can OnTriggerEnter be called multiple times? Does this check handle it?

Reply:
yes, it will be called for each collider entering, but this is expected, so there will be multiple entries (of the same object) in the hoveredTargets list, but every collider which enters does also have to leave.

Reply:
Can this be optimized, I think the interactors already have a list of hovered targets, maybe the whole logic could be handled from there.

Reply:
I don't think we have to optimize anything right now.

Reply:
Agree, but we should at some point.